service/Abstractions/Pipeline/DataPipeline.cs

```diff
@@ -149,12 +149,19 @@ public void MarkProcessedBy(IPipelineStepHandler handler)
         /// <param name="text">Text to store for the end user</param>
         public void Log(IPipelineStepHandler handler, string text)
         {
-            if (this.LogEntries == null)
-            {
-                this.LogEntries = new List<PipelineLogEntry>();
-            }
+            this.Log(handler.StepName, text);
+        }
 
-            this.LogEntries.Add(new PipelineLogEntry(source: handler.StepName, text: text));
+        /// <summary>
+        /// Add a new log entry, with some important information for the end user.
+        /// DO NOT STORE PII OR SECRETS here.
+        /// </summary>
+        /// <param name="handlerStepName">Step name of the handler sending the information to log</param>
+        /// <param name="text">Text to store for the end user</param>
+        public void Log(string handlerStepName, string text)
+        {
+            this.LogEntries ??= new List<PipelineLogEntry>();
+            this.LogEntries.Add(new PipelineLogEntry(source: handlerStepName, text: text));
```

Comment:
I would remove this, e.g. decoders should not know anything about the pipeline and it's internal logging. If needed, we can include log data in FileContent, or return log data as a separate object.

Reply:
I think that adding adding log data in `FileContent` could be a good solution. As we need this for logging the reason of  skipping the file, it could be a simple `SkippingReason` string property. Or do you have in mind something more structured?